src/components/node-list/node-list-tree.js

```diff
@@ -0,0 +1,207 @@
+import React, { useState, useEffect } from 'react';
+import { connect } from 'react-redux';
+
+import { makeStyles, withStyles } from '@material-ui/core/styles';
+import TreeView from '@material-ui/lab/TreeView';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+
+import {
+  toggleModularPipelineActive,
+  toggleModularPipelineFilter,
+} from '../../actions/modular-pipelines';
+import { toggleTypeDisabled } from '../../actions/node-type';
+import { getNodeTypes, getNodeTypeIDs } from '../../selectors/node-types';
+import {
+  getModularPipelineIDs,
+  getModularPipelineData,
+} from '../../selectors/modular-pipelines';
+import {
+  getNestedModularPipelines,
+  getFilteredTreeItems,
+} from './node-list-items';
+import {
+  getGroupedNodes,
+  getNodeSelected,
+  getNodeModularPipelines,
+} from '../../selectors/nodes';
+import { loadNodeData } from '../../actions/nodes';
+import NodeListTreeItem from './node-list-tree-item';
+
+// please note that this setup is unique for initialization of the material-ui tree,
+// and setup is only used here and not anywhere else in the app.
+const useStyles = makeStyles({
+  root: {
+    height: 110,
+    flexGrow: 1,
+    maxWidth: 400,
+  },
+});
+
+const StyledTreeView = withStyles({
+  root: {
+    padding: '0 0 0 20px',
+  },
+})(TreeView);
+
+const isModularPipelineType = (type) => type === 'modularPipeline';
+
+const TreeListProvider = ({
+  nodes,
+  nodeSelected,
+  onToggleNodeSelected,
+  searchValue,
+  modularPipelines,
+  modularPipelineIds,
+  nodeModularPipelines,
+  onItemChange,
+  onItemMouseEnter,
+  onItemMouseLeave,
+  nodeTypeIDs,
+  searching,
+}) => {
+  const classes = useStyles();
+
+  const [expandedPipelines, setExpandedPipelines] = useState([]);
+
+  useEffect(() => {
+    const filteredTreeItems =
+      searchValue !== ''
+        ? getFilteredTreeItems({
+            nodes,
+            modularPipelines,
+            nodeSelected,
+            searchValue,
+            modularPipelineIds,
+            nodeModularPipelines,
+            nodeTypeIDs,
+          })
+        : [];
+
+    let expandedModularPipelines = [];
+
+    searchValue !== '' &&
```

Comment:
How come there is a conditional here? Should this be inside an if statement instead? Same goes for the above lines in this `useEffect`?

Reply:
This is to detect the cases for empty searchValue in the search bar - this is also demonstrated in our legacy tests throughout the code base in detecting the case of empty value in the search bar

Reply:
Yeah I get what you mean, I just meant it should probably be inside an `if (...) { ... } ` otherwise it's not easy to see that it's a conditional?